chia/wallet/did_wallet/did_wallet.py

```diff
@@ -306,7 +306,7 @@ async def select_coins(
         exclude: Optional[List[Coin]] = None,
         min_coin_amount: Optional[uint64] = None,
         max_coin_amount: Optional[uint64] = None,
-    ) -> Optional[Set[Coin]]:
+    ) -> Set[Coin]:
```

Comment:
I think DID should be one of the wallets that raises on this method, it looks like right here the guaranteed behavior is just to return your DID every time.  And amount passed to this > 1 would fail I think.  Maybe out of scope of this PR though.

Reply:
I don't think I follow. In this patch I make this function throw instead of returning `None`. The only condition where it used to return `None` was if the `amount` > `spendable_amount`. It falls back on the generic `select_coins()` function though, but is it the case that there will always only be a single coin in a DID wallet?